lib/firestore/delete.js

```diff
@@ -280,18 +273,120 @@ FirestoreDelete.progressBar = new ProgressBar("Deleted :current docs (:rate docs
  */
 FirestoreDelete.prototype._recursiveBatchDelete = function() {
   var self = this;
-  return this._getDescendantBatch(this.allDescendants, this.batchSize).then(function(docs) {
-    if (docs.length <= 0) {
-      return Promise.resolve();
+
+  // Tunable deletion parameters
+  var readBatchSize = 7500;
+  var deleteBatchSize = 250;
+  var maxPendingDeletes = 15;
+  var maxQueueSize = deleteBatchSize * maxPendingDeletes * 2;
+
+  // All temporary variables for the deletion queue.
+  var queue = [];
+  var numPendingDeletes = 0;
+  var pagesRemaining = true;
+  var pageIncoming = false;
+  var lastDocName;
+
+  var failures = [];
+  var retried = {};
+
+  var queueLoop = function() {
```

Comment:
If you make this an arrow function, then you don't need `self`.

Reply:
We can't use arrow functions in those codebase, unfortunately.